packages/interactivity/src/init.js

```diff
@@ -30,17 +30,47 @@ function yieldToMain() {
 
 // Initialize the router with the initial DOM.
 export const init = async () => {
+	const pendingNodes = new Set();
+
+	const intersectionObserver = new window.IntersectionObserver(
```

Comment:
If there are no `nodes` below, then this wouldn't need to be constructed in the first place.